library/testing/PseudoUnitTest.h

```diff
@@ -0,0 +1,150 @@
+#ifndef Test_h
+#define Test_h
+
+#include <functional>
+#include <iostream>
+#include <sstream>
+
+/** Helper to perform multiple checks within the same `ctest` test.
+ * Checks are performed using the various overloads of `operator()`
+ * and their results are logged and tracked so that the `result()` method
+ * will return `EXIT_FAILURE` if any of them failed (and `EXIT_SUCCESS` otherwise).
+ */
+class PseudoUnitTest
+{
+public:
+  /** test a boolean condition */
+  void operator()(const std::string& label, bool condition)
+  {
+    this->record(condition, label);
+  }
+
+  /** test the execution of a function */
+  void operator()(const std::string& label, std::function<void()> function)
+  {
+    this->testFunction<Dummy>(label, function);
+  }
+
+  /** test the execution of a function expecting a given exception */
+  template<typename E>
+  void expect(const std::string& label, std::function<void()> function)
```

Comment:
Not super important, but `std::function` is expensive, using template allow the compiler to optimize away the function and inline it. 
```suggestion
  template<typename E, typename F>
  void expect(const std::string& label, F function)
```